src/Microsoft.SqlTools.ServiceLayer/ObjectExplorer/SmoModel/SmoWrapper.cs

```diff
@@ -38,5 +53,21 @@ public virtual void OpenConnection(SmoObjectBase smoObj)
                 sqlObj.ExecutionManager.ConnectionContext.Connect();
             }
         }
+
+        /// <summary>
+        /// Updates access token on the connection context of <paramref name="smoObj"/> instance.
+        /// </summary>
+        /// <param name="smoObj">SMO Object containing connection context.</param>
+        /// <param name="accessToken">Access token</param>
+        public virtual void UpdateAccessToken(SmoObjectBase smoObj, string accessToken)
```

Comment:
Seems like a good candidate for an extension function (`UpdateAccessToken(this SmoObjectBase smoObj, string accessToken)`)